src/Step/Build/CompressStaticContent.php

```diff
@@ -76,18 +80,26 @@ public function __construct(
      */
     public function execute()
     {
-        if ($this->flagManager->exists(FlagManager::FLAG_STATIC_CONTENT_DEPLOY_IN_BUILD)) {
+        try {
+            if (!$this->flagManager->exists(FlagManager::FLAG_STATIC_CONTENT_DEPLOY_IN_BUILD)) {
+                $this->logger->info(
+                    'Skipping build-time static content compression because static content deployment has not happened.'
+                );
+
+                return;
+            }
+
             $this->compressor->process(
                 $this->stageConfig->get(BuildInterface::VAR_SCD_COMPRESSION_LEVEL),
                 $this->stageConfig->get(BuildInterface::VAR_SCD_COMPRESSION_TIMEOUT),
                 $this->stageConfig->get(BuildInterface::VAR_VERBOSE_COMMANDS)
             );
-        } else {
-            $this->logger->info(
-                'Skipping build-time static content compression because static content deployment has not happened.'
-            );
-
-            return;
+        } catch (ShellException $e) {
+            throw new StepException($e->getMessage(), Error::BUILD_SCD_COMPRESSION_FAILED, $e);
+        } catch (UtilityException $e) {
+            throw new StepException($e->getMessage(), Error::BUILD_UTILITY_NOT_FOUND, $e);
+        } catch (\Exception $e) {
+            throw new StepException($e->getMessage(), $e->getCode(), $e);
```

Comment:
Is it necessary to convert \Exception to StepException here? as I did not see this conversion in other steps.

Reply:
Yes, as `Step` classes can throw only `StepException` it's defined in the `StepInterface`
If somewhere in the `Step` classes are throwing a different exception it should be fixed.

Reply:
There are a lot of places where \Exception is not caught:
- src/Step/Build/ApplyPatches.php
- src/Step/Build/BackupData/StaticContent.php
- src/Step/Build/BackupData/WritableDirectories.php
...

Reply:
Not all steps need to catch `\Exception` if all possible exceptions are already caught.
I only catching `\Exception` where need to catch several different exceptions and pass theirs error code to the StepException.

Reply:
Yes, in some steps like `WritableDirectories` I missed to catch all exception, will fix it